src/client/Microsoft.Identity.Client/Platforms/Features/OpenTelemetry/OtelInstrumentation.cs

```diff
@@ -0,0 +1,182 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Cache;
+using Microsoft.Identity.Client.TelemetryCore;
+using Microsoft.Identity.Client.TelemetryCore.OpenTelemetry;
+using System.Diagnostics.Metrics;
+using Microsoft.Identity.Client.Internal;
+
+namespace Microsoft.Identity.Client.Platforms.Features.OpenTelemetry
+{
+    /// <summary>
+    /// Class to hold the OpenTelemetry objects used by MSAL.
+    /// </summary>
+    internal class OtelInstrumentation : IOtelInstrumentation
+    {
+        /// <summary>
+        /// Constant to hold the name of the Meter.
+        /// </summary>
+        public const string MeterName = "MicrosoftIdentityClient_Common_Meter";
+
+        /// <summary>
+        /// Constant to holt the name of the ActivitySource.
+        /// </summary>
+        public const string ActivitySourceName = "MicrosoftIdentityClient_Activity";
+
+        private const string SuccessCounterName = "MsalSuccess";
+        private const string FailedCounterName = "MsalFailed";
+        private const string TotalDurationHistogramName = "MsalTotalDuration.1A";
+        private const string TotalDurationInUsHistogramName = "MsalTotalDurationInUs.1B";
+        private const string DurationInCacheHistogramName = "MsalDurationInCache.1A";
+        private const string DurationInHttpHistogramName = "MsalDurationInHttp.1A";
+
+        /// <summary>
+        /// Meter to hold the MSAL metrics.
+        /// </summary>
+        internal static readonly Meter Meter = new Meter(MeterName, "1.0.0");
+
+        /// <summary>
+        /// ActivitySource to hold the MSAL activities.
+        /// </summary>
+        internal static readonly ActivitySource s_acquireTokenActivity = new ActivitySource(ActivitySourceName, "1.0.0");
```

Comment:
I assume this is a temporary collection, which gets cleared out once the exporter exports the data? This doesn't grow indefinitely? Can a limit be set on the size of this or does it depend on the export frequency set?
Also to my previous point, if platform supports Otel but it's not enabled, this will just get populated but never cleared, right?

Reply:
Also same question about counters and histogram objects below - when are they cleared.

Reply:
Based on sync - the exporter clears the collection after exporting data.